File: mobile/src/main/java/tr/xip/prayertimes/api/objects/PrayerTime.java
package tr.xip.prayertimes.api.objects;

import android.graphics.drawable.Drawable;

import tr.xip.prayertimes.R;
import tr.xip.prayertimes.Utils;

import static tr.xip.prayertimes.ui.app.NamazVakitleriApplication.getContext;

public class PrayerTime {
    private long time;
    private Type type;

    public PrayerTime(Type type, long time) {
        this.time = time;
        this.type = type;
    }

    public PrayerTime(Type type, String date, String time) {
        this.time = Utils.getTimestampFromDateTime(date, time);
        this.type = type;
    }

    public long getTime() {
        return time;
    }

    public String getTitle() {
        switch (type) {
            case FAJR:
                return getContext().getString(R.string.prayer_time_fajr);
            case SUNRISE:
                return getContext().getString(R.string.prayer_time_sunrise);
            case DHUHR:
                return getContext().getString(R.string.prayer_time_dhuhr);
            case ASR:
                return getContext().getString(R.string.prayer_time_asr);
            case MAGHRIB:
                return getContext().getString(R.string.prayer_time_maghrib);
            case ISHA:
                return getContext().getString(R.string.prayer_time_isha);
            default:
                return "?";
        }
    }

    public Drawable getSmallImage() {
        switch (type) {
            case FAJR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_fajr_small);
            case SUNRISE:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_sunrise_small);
            case DHUHR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_dhuhr_small);
            case ASR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_asr_small);
            case MAGHRIB:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_maghrib_small);
            case ISHA:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_isha_small);
            default:
                return null;
        }
    }

    public Drawable getLargeImage() {
        switch (type) {
            case FAJR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_fajr_large);
            case SUNRISE:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_sunrise_large);
            case DHUHR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_dhuhr_large);
            case ASR:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_asr_large);
            case MAGHRIB:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_maghrib_large);
            case ISHA:
                return getContext().getResources().getDrawable(R.drawable.prayer_time_isha_large);
            default:
                return null;
        }
    }

    public enum Type {
        FAJR, SUNRISE, DHUHR, ASR, MAGHRIB, ISHA
    }
}

File: mobile/src/main/kotlin/tr/xip/prayertimes/model/PrayerTime.kt
package tr.xip.prayertimes.model

import android.graphics.drawable.Drawable

import java.io.Serializable

import tr.xip.prayertimes.R
import tr.xip.prayertimes.util.getContext
import tr.xip.prayertimes.util.getTimestampFromDateTime

class PrayerTime : Serializable {
    var time: Long = 0
    var type: Type? = null

    constructor(type: PrayerTime.Type, time: Long) {
        this.time = time
        this.type = type
    }

    constructor(type: PrayerTime.Type, date: String, time: String) {
        this.time = getTimestampFromDateTime(date, time)
        this.type = type
    }

    val title: String
        get() {
            when (type) {
                Type.FAJR -> return getContext().getString(R.string.prayer_time_fajr)
                Type.SUNRISE -> return getContext().getString(R.string.prayer_time_sunrise)
                Type.DHUHR -> return getContext().getString(R.string.prayer_time_dhuhr)
                Type.ASR -> return getContext().getString(R.string.prayer_time_asr)
                Type.MAGHRIB -> return getContext().getString(R.string.prayer_time_maghrib)
                Type.ISHA -> return getContext().getString(R.string.prayer_time_isha)
                else -> return "?"
            }
        }

    val smallImage: Drawable?
        get() {
            when (type) {
                Type.FAJR -> return getContext().resources.getDrawable(R.drawable.prayer_time_fajr_small, null)
                Type.SUNRISE -> return getContext().resources.getDrawable(R.drawable.prayer_time_sunrise_small, null)
                Type.DHUHR -> return getContext().resources.getDrawable(R.drawable.prayer_time_dhuhr_small, null)
                Type.ASR -> return getContext().resources.getDrawable(R.drawable.prayer_time_asr_small, null)
                Type.MAGHRIB -> return getContext().resources.getDrawable(R.drawable.prayer_time_maghrib_small, null)
                Type.ISHA -> return getContext().resources.getDrawable(R.drawable.prayer_time_isha_small, null)
                else -> return null
            }
        }

    val largeImage: Drawable?
        get() {
            when (type) {
                Type.FAJR -> return getContext().resources.getDrawable(R.drawable.prayer_time_fajr_large, null)
                Type.SUNRISE -> return getContext().resources.getDrawable(R.drawable.prayer_time_sunrise_large, null)
                Type.DHUHR -> return getContext().resources.getDrawable(R.drawable.prayer_time_dhuhr_large, null)
                Type.ASR -> return getContext().resources.getDrawable(R.drawable.prayer_time_asr_large, null)
                Type.MAGHRIB -> return getContext().resources.getDrawable(R.drawable.prayer_time_maghrib_large, null)
                Type.ISHA -> return getContext().resources.getDrawable(R.drawable.prayer_time_isha_large, null)
                else -> return null
            }
        }

    enum class Type {
        FAJR, SUNRISE, DHUHR, ASR, MAGHRIB, ISHA
    }
}
